star/private/diffusion_support.f90

```diff
@@ -1666,6 +1681,132 @@ subroutine lam_SM(nc,m,rho,T,Z,nd,lam_eff,kappa)
+        Kfac = boltzm*T/(ni_sum*Zp6bar*pow(Zbar,0.6d0)*Docp)
+        do i = 1,nc
+           do j = 1,nc
+              Kdiff(i,j) = nd(i)*nd(j)*pow(Z(i)*Z(j),0.6d0)*Kfac
+           end do
+        end do
+
+      end subroutine get_CBF_coeffs
+      
+      subroutine get_Dstar_OCP(Gamma,kappa,DstarOCP)
+        real(dp), intent(in) :: Gamma, kappa
+        real(dp), intent(out) :: DstarOCP
+
+        real(dp) :: a0, a1, a2, b0, b1, b2, c0, c1, c2, c3
+        real(dp) :: Ak, Bk, Ck
+
+        a0 = 1.55973d0
+        a1 = 1.10941d0
+        a2 = 1.36909d0
+        
+        b0 = 0.0070782d0
+        b1 = 0.80499d0
+        b2 = 4.53523d0
+        
+        c0 = 2.20689d0
+        c1 = 1.351594d0
+        c2 = 1.57138d0
+        c3 = 3.34187d0
+        
+        Ak = sqrt(pi/3d0)*(a0 + a1*pow(kappa,a2))
+        Bk = b0*exp(-b1*pow(kappa,b2))
+        Ck = c0 + c1*bitsafe_erf_fit(c2*pow(kappa,c3))
+
+        ! Eqn (5)
+        DstarOCP = sqrt(pi/3d0)*Ak*pow(Gamma,-2.5d0)*exp(-Bk*Gamma)/log(1d0 + Ck*pow(Gamma,-1.5d0)/sqrt(3d0))
+
+      end subroutine get_Dstar_OCP
+      
+      ! Screening Length for CBF diffusion coefficients.
+      ! Only electron screening, accounts for potentially relativistic electron degeneracy.
+      subroutine kappa_CBF(nc,m,rho,T,Abar,Zbar,nd,omegap,ai,lam_e,kappa)
+        integer, intent(in) :: nc, m
+        real(dp), intent(in) :: rho, T, Abar, Zbar
+        real(dp), dimension(:), intent(in) :: nd ! m. number densities of all species
+        real(dp), intent(out) :: omegap, ai, lam_e
+        real(dp), intent(out) :: kappa ! ion separation relative to electron screening length.
+
+        real(dp) :: tiny_n, ne, EF, kF, lam_e_SM, lam_e_rel, ni_tot
+        integer :: i
+
+        tiny_n = 1d-20 ! g/cc
+        ne = nd(m) ! electron number density
+        ! Electron Fermi energy
+        EF = (hbar*hbar*pow(3d0*pi*pi*ne,2d0/3d0))/(2d0*me)
+        ! Electron screening length accounting for degeneracy correction
+        lam_e_SM = pow(pi4*qe*qe*ne/sqrt(pow2(boltzm*T) + pow2((2d0/3d0)*EF)),-0.5d0)
+
+        if (rho < 1d6) then
+           lam_e = lam_e_SM
+        else ! calculate and use relativistic screening length if rho > 1d6
+           kF = pow(3d0*pi*pi*ne,one_third) ! electron Fermi momentum
+           lam_e_rel = 1d0/(2d0*kF*sqrt(fine/pi)) ! fine = fine structure constant ~1/137
+           lam_e = min(lam_e_SM,lam_e_rel) ! min for continuity
+        end if
 
+        ! Compute kappa
+        ni_tot = 0d0
+        do i = 1,nc
+           ni_tot = ni_tot + nd(i)
+        end do
+        omegap = sqrt(pi4*ni_tot*Zbar*Zbar*qe*qe/(Abar*amu)) ! plasma frequency
+        ai = pow(3d0/(pi4*ni_tot),one_third)
+        kappa = ai/lam_e
+
+      end subroutine kappa_CBF
+
+      ! crlibm does not include error function, so implement an
+      ! erf fit good to 2.5e-5 (Abramowitz and Stegun 1964)
+      ! in terms of other bitsafe functions.
+      real(dp) function bitsafe_erf_fit(x)
```

Comment:
Do we want to somehow implement the bit-safe error function in a more basic module (e.g. `math`) so it's available to other parts of MESA and in `run_star_extras.f90`?

Reply:
That sounds potentially reasonable, but perhaps we can raise this as a rainy-day issue after merging this? Also, is 2.5e-5 accurate enough to use as a general drop-in replacement for `erf`? It's certainly fine for this particular application, but I'm not sure I'd want to add this to math in a way that users might not realize they're not using an approximation.